pkg/composer/serviceparser/build.go

```diff
@@ -86,6 +87,20 @@ func parseBuildConfig(c *types.BuildConfig, project *types.Project, imageName st
 		}
 	}
 
+	if c.DockerfileInline != "" {
+		// if DockerfileInline is specified, write it to a temporary file
+		// and use -f flag to use that docker file with project's ctxdir
+		tmpFile, err := os.CreateTemp("", fmt.Sprintf("%s.Dockerfile", imageName))
+		if err != nil {
+			return nil, fmt.Errorf("failed to create temp file for DockerfileInline: %w", err)
+		}
+		defer tmpFile.Close()
+		if _, err := tmpFile.Write([]byte(c.DockerfileInline)); err != nil {
+			return nil, fmt.Errorf("failed to write DockerfileInline: %w", err)
+		}
+		b.BuildArgs = append(b.BuildArgs, "-f="+tmpFile.Name())
+	}
+
```

Comment:
The tmp file has to be removed on completion of the build?

Reply:
Hmm, at the moment, it does not clear it up. We are creating this file at the "service parser" phase, the actual build command just gets back the modified `BuildArgs`. I will take a look and see if I can clear up the `tmp` file.

Reply:
It seems like I can store the `dockerfile_inline` value in the `Build` struct, and then during the build phase, I can create the temporary file, use it for the build, and then clear it up. 

```go
type Build struct {
	Force     bool     // force build even if already present
	BuildArgs []string // {"-t", "example.com/foo", "--target", "foo", "/path/to/ctx"}
	// something like follows
	DockerfileInline string
       // TODO: call BuildKit API directly without executing `nerdctl build`
}
```

Do you have any objections to putting it there?

Reply:
No objection

Reply:
Done.